src/sorting/mod.rs

```diff
@@ -50,27 +52,35 @@ pub use self::sleep_sort::sleep_sort;
 pub use self::stooge_sort::stooge_sort;
 pub use self::tim_sort::tim_sort;
 
+#[cfg(test)]
 use std::cmp;
 
-pub fn is_sorted<T>(arr: &[T]) -> bool
+#[cfg(test)]
+pub fn have_same_elements<T>(a: &[T], b: &[T]) -> bool
 where
-    T: cmp::PartialOrd,
+    T: cmp::PartialOrd, // If HashSet is used
+                        // T: cmp::PartialOrd + cmp::Eq + std::hash::Hash
 {
-    if arr.is_empty() {
-        return true;
-    }
-
-    let mut prev = &arr[0];
+    match a.len() == b.len() {
+        true => {
+            // This is O(n^2) but performs better on smaller data sizes
+            b.iter().all(|item| a.contains(item))
 
-    for item in arr.iter().skip(1) {
-        if prev > item {
-            return false;
+            // This is O(n), performs well on larger data sizes
+            // let set_a: HashSet<&T> = a.iter().collect();
+            // let set_b: HashSet<&T> = b.iter().collect();
+            // set_a == set_b
```

Comment:
I'd use sets to avoid quadratic time if it's run with bigger inputs. On small arrays, it runs fast in any case, right?

Reply:
Yes, small arrays it runs quick, but linear is quicker with smaller sets. Here's some numbers to help the choice:

compare_linearly(10) = true, elapsed = 6 μs
compare_using_hashset(10) = true, elapsed = 1 μs
compare_linearly(100) = true, elapsed = 9 μs
compare_using_hashset(100) = true, elapsed = 42 μs
compare_linearly(1000) = true, elapsed = 63 μs
compare_using_hashset(1000) = true, elapsed = 4_243 μs
compare_linearly(10000) = true, elapsed = 621 μs
compare_using_hashset(10000) = true, elapsed = 375_390 μs
compare_linearly(100000) = true, elapsed = 6_123 μs
compare_using_hashset(100000) = true, elapsed = 67_596_083 μs

Reply:
So up to 100_000 elements O(n2) algorithm is faster than O(n)?

Reply:
The data you provided suggests that `compare_linearly`is O(n) and `compare_using_hashset` is O(n2)

Reply:
Oops - I think I am printing the function names reversed - let me fix that

Reply:
Fixed. Now you can see the correct function names with correct numbers (my bad: the println!() text was reversed somehow)

Reply:
That's what I thought, now the data matches my expectations. Sub-millisecond values don't matter, they are all way too fast to notice, and even at 100 elements hash sets are already faster.